pymc/sampling/forward.py

```diff
@@ -821,6 +824,7 @@ def sample_posterior_predictive(
         vars_ = model.observed_RVs + observed_dependent_deterministics(model)
         if observed_data is not None:
             vars_ += [model[x] for x in observed_data if x in model and x not in vars_]
+            vars_ += observed_dependent_deterministics(model, vars_)
```

Comment:
This is going to duplicate deterministics of an observed variable (in case there's a mix of observed model variables and implied observed variables from the idata. the `+ observed_dependent_determininstics)` should be called only once after the if branch?